.github/workflows/cmake.yml

```diff
@@ -1,13 +1,6 @@
 name: CI build
 
 on:
-  push:
```

Comment:
actually, was this change intentional?

Reply:
Yes. This PR changes the way GitHub workflows are triggered. Before, cmake.yml was triggered by push to default and PR HEADs based on main _and_ as a reusable (callable) workflow on release.

Now, cmake.yml is only called by other workflows, and test-deployments.yml takes over calling cmake.yml on push to default and PR HEADs based on main.

Reply:
Assuming I understand you, this feels inverted to me. i would think the flow is:

- build the artifacts
- upload the artifacts to the job
- kick off other 'test' jobs that download the artifact they want to test/verifiy
- mark the job successful/failed

having something like "test-deployments" initiate the build -- seems wrong. Maybe I'm not understanding you?

Reply:
Correct. That is the flow. The Test deployments workflow runs on push to default and PR HEADs based on default. We can name it whatever.